File: backend/stathis/src/main/java/edu/cit/stathis/vitals/entity/VitalSigns.java
package edu.cit.stathis.vitals.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "vital_signs")
public class VitalSigns {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "student_id", nullable = false)
    private Long studentId;

    @Column(name = "classroom_id", nullable = false)
    private Long classroomId;

    @Column(name = "task_id", nullable = false)
    private Long taskId;

    @Column(name = "heart_rate")
    private Integer heartRate;

    @Column(name = "oxygen_saturation")
    private Integer oxygenSaturation;

    @Column(name = "timestamp", nullable = false)
    private LocalDateTime timestamp;

    @Column(name = "is_pre_activity", nullable = false)
    private Boolean isPreActivity;

    @Column(name = "is_post_activity", nullable = false)
    private Boolean isPostActivity;
} 

File: mobile/stathis-mobile/app/src/main/java/citu/edu/stathis/mobile/features/vitals/domain/model/VitalSigns.kt
package citu.edu.stathis.mobile.features.vitals.domain.model

import java.time.LocalDateTime

data class VitalSigns(
    val id: String? = null,
    val userId: String,
    val systolicBP: Int,
    val diastolicBP: Int,
    val heartRate: Int,
    val respirationRate: Int,
    val temperature: Float,
    val oxygenSaturation: Float,
    val timestamp: LocalDateTime = LocalDateTime.now(),
    val deviceName: String? = null
)